service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/azure/kubernetesNamespace/ControlledAzureKubernetesNamespaceResource.java

```diff
@@ -0,0 +1,433 @@
+
+  @JsonCreator
+  public ControlledAzureKubernetesNamespaceResource(
+      @JsonProperty("wsmResourceFields") WsmResourceFields resourceFields,
+      @JsonProperty("wsmControlledResourceFields")
+          WsmControlledResourceFields controlledResourceFields,
+      @JsonProperty("kubernetesNamespace") String kubernetesNamespace,
+      @JsonProperty("kubernetesServiceAccount") String kubernetesServiceAccount,
+      @JsonProperty("managedIdentity") UUID managedIdentity,
+      @JsonProperty("databases") Set<UUID> databases) {
+    super(resourceFields, controlledResourceFields);
+    this.kubernetesNamespace = kubernetesNamespace;
+    this.kubernetesServiceAccount = kubernetesServiceAccount;
+    this.managedIdentity = managedIdentity;
+    this.databases = databases;
+    validate();
+  }
+
+  // Constructor for the builder
+  private ControlledAzureKubernetesNamespaceResource(
+      ControlledResourceFields common,
+      String kubernetesNamespace,
+      String kubernetesServiceAccount,
+      UUID managedIdentity,
+      Set<UUID> databases) {
+    super(common);
+    this.kubernetesNamespace = kubernetesNamespace;
+    this.kubernetesServiceAccount = kubernetesServiceAccount;
+    this.managedIdentity = managedIdentity;
+    this.databases = databases;
+    validate();
+  }
+
+  @Override
+  public WsmResourceFields getWsmResourceFields() {
+    return super.getWsmResourceFields();
+  }
+
+  @Override
+  public WsmControlledResourceFields getWsmControlledResourceFields() {
+    return super.getWsmControlledResourceFields();
+  }
+
+  public String getKubernetesNamespace() {
+    return kubernetesNamespace;
+  }
+
+  public String getKubernetesServiceAccount() {
+    return kubernetesServiceAccount;
+  }
+
+  public UUID getManagedIdentity() {
+    return managedIdentity;
+  }
+
+  public Set<UUID> getDatabases() {
+    return databases;
+  }
+
+  @Override
+  @JsonIgnore
+  public Optional<UniquenessCheckAttributes> getUniquenessCheckAttributes() {
+    return Optional.of(
+        new UniquenessCheckAttributes()
+            .uniquenessScope(UniquenessScope.WORKSPACE)
+            .addParameter("kubernetesNamespace", getKubernetesNamespace()));
+  }
+
+  @Override
+  public void addCreateSteps(
+      CreateControlledResourceFlight flight,
+      String petSaEmail,
+      AuthenticatedUserRequest userRequest,
+      FlightBeanBag flightBeanBag) {
+    RetryRule cloudRetry = new RetryRuleNone();
+
+    getCreateSteps(flightBeanBag).forEach(step -> flight.addStep(step, cloudRetry));
+  }
+
+  @VisibleForTesting
+  List<Step> getCreateSteps(FlightBeanBag flightBeanBag) {
+    /*
+    Flight plan:
+    > ensure the kubernetes namespace does not already exist
+    > create the kubernetes namespace
+
+    if a managed identity is provided or this is a private resource:
+    > get the correct managed identity (either from workspace or pet)
+    > create the federated credentials (both KSA and federated credentials in managed identity)
```

Comment:
I have no intuition for what "federated" means here. Can you explain what it means for a credential to be federated?